### PR TITLE
Normalized heatmap losses

### PR DESCRIPTION
**Pull request by themattinthehatt**
* normalize ground truth heatmaps
* normalize heatmap output of model with softmax
* implement KL and JS heatmap losses with kornia
* filter out low-confidence bodyparts when computing unimodal loss

### CODE REVIEW

File: lightning_pose/losses/losses.py [hunk 1]
```diff
@@ -402,12 +463,11 @@ class UnimodalLoss(Loss):
...
         self,
-        loss_name: Literal["unimodal_mse", "unimodal_wasserstein"],
+        loss_name: Literal["unimodal_mse", "unimodal_kl", "unimodal_js"],
         original_image_height: int,
         original_image_width: int,
         downsampled_image_height: int,
         downsampled_image_width: int,
-        reach: Union[float, str] = "none",
         data_module: Optional[Union[BaseDataModule, UnlabeledDataModule]] = None,
         epsilon: float = 0.0,
```
Comment (danbider) [inline lightning_pose/losses/losses.py:472]: modify name?

File: lightning_pose/models/heatmap_tracker.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def __init__(
         self.upsampling_layers = self.make_upsampling_layers()
         self.initialize_upsampling_layers()
         self.output_shape = output_shape
-        self.temperature = torch.tensor(100, device=self.device)  # soft argmax temp
+        self.temperature = torch.tensor(1000., device=self.device)  # soft argmax temp
```
Comment (danbider) [inline lightning_pose/models/heatmap_tracker.py:84]: needs to be tested -- varies as the number of pixels in a heatmap
add a comment saying to be careful here.

File: lightning_pose/models/heatmap_tracker.py [hunk 3]
```diff
@@ -191,14 +201,8 @@ def forward(
...
-        # B = heatmaps.shape[0]
-        # valid_probability_heatmaps = self.softmax(
-        #     heatmaps.reshape(B, self.num_keypoints, -1)
-        # )
-        # valid_probability_heatmaps = valid_probability_heatmaps.reshape(
-        #     B, self.num_keypoints, self.output_shape[0], self.output_shape[1]
-        # )
-        return heatmaps
+        # return heatmaps
+        return spatial_softmax2d(heatmaps, temperature=torch.tensor([1.0]))
```
Comment (danbider) [inline lightning_pose/models/heatmap_tracker.py:205]: comment should be always 1.00 std softmax

File: scripts/configs/losses/losses_params.yaml [hunk 4]
```diff
@@ -70,11 +70,12 @@ temporal:
 unimodal_mse:
   # weight in front of unimodal loss
   log_weight: 6.5
+  # ignore heatmaps with a max prob less than epsilon (so loss is not applied to occlusions)
+  esilon: 0.0
```
Comment (danbider) [inline scripts/configs/losses/losses_params.yaml:74]: typo

File: scripts/configs/losses/losses_params.yaml [hunk 5]
```diff
@@ -70,11 +70,12 @@ temporal:
...
+# loss = kl loss between generated heatmap and ideal gaussian heatmap with same center as generated heatmap
+unimodal_kl:
   # weight in front of unimodal loss
-  log_weight: 0.0
-
-  # todo
-  reach: none
+  log_weight: 6.5
+  # ignore heatmaps with a max prob less than epsilon (so loss is not applied to occlusions)
+  esilon: 0.0
```
Comment (danbider) [inline scripts/configs/losses/losses_params.yaml:81]: typo

**Review by danbider — CHANGES_REQUESTED** [review]

**Review by danbider — APPROVED** [review]

**Comment by danbider** [comment]
nice -- merging.

File: lightning_pose/losses/losses.py [hunk 6]
```diff
@@ -402,12 +463,11 @@ class UnimodalLoss(Loss):
     @typechecked
```
[the remark this answers is not in the document]
Comment (Gam3356) [reply lightning_pose/losses/losses.py]: Please check The IP
